Fabric.Authorization.Persistence.SqlServer/Stores/EDWStore.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Linq;
+using Fabric.Authorization.Domain.Models.EDW;
+using Fabric.Authorization.Domain.Stores;
+using Fabric.Authorization.Persistence.SqlServer.Stores.EDW;
+using Microsoft.EntityFrameworkCore;
+
+namespace Fabric.Authorization.Persistence.SqlServer.Stores
+{
+    public class EDWStore : /*SqlServerBaseStore,TODO EDWServerBaseStore?*/ IEDWStore
+    {
+        private readonly ISecurityContext securityContext;
+
+        public EDWStore(ISecurityContext securityContext)
+        {
+            this.securityContext = securityContext ??
+                    throw new ArgumentNullException(nameof(securityContext));
+        }
+
+        public void AddIdentitiesToRole(string[] identities, string roleName)
+        {
+            var role = this.GetRolesByName(roleName).SingleOrDefault();
+            this.AddIdentitiesToRole(identities, role.Id);
+        }
+
+
+        public void RemoveIdentitiesFromRole(string[] identities, string roleName)
+        {
+            var role = this.GetRolesByName(roleName).SingleOrDefault();
+            this.RemoveIdentitiesFromRole(identities, role.Id);
+        }
+
+
+        private IQueryable<EDWRole> GetRolesByName(string roleName)
+        {            
+            return this.securityContext.EDWRoles.Where(role => role.Name == roleName);
+        }
+
+        private void AddIdentitiesToRole(string[] identityNames, int roleId)
+        {
+            using (var context = this.securityContext.CreateContext())
+            {
+                foreach (string identityName in identityNames)
+                {
+                    if (!context.EDWIdentities.Any(identity => identity.Name == identityName))
+                    {
+                        context.EDWIdentities.Add(new EDWIdentity { Name = identityName });
+                    }
+                }
+
+                context.SaveChanges();
+
+                var roleToAdd = context.EDWRoles.FirstOrDefault(role => role.Id == roleId);
+
+                if (roleToAdd == null)
+                {
+                    throw new ArgumentException("Role not found.");
```

Comment:
I think a more informative error message here would be helpful. Since the app is using two context and each has a concept of role, we should at least let people know which role we are looking from which context.

Reply:
Should we also consider some sort of partial success, or perhaps wrap the whole thing in a transaction? What it looks like can happen now is, the auth database gets updated, but this update fails, and now we are inconsistent...